pkg/archive/changes.go

```diff
@@ -220,8 +220,8 @@ func (info *FileInfo) addChanges(oldInfo *FileInfo, changes *[]Change) {
 				oldStat.Gid() != newStat.Gid() ||
 				oldStat.Rdev() != newStat.Rdev() ||
 				// Don't look at size for dirs, its not a good measure of change
-				(oldStat.Size() != newStat.Size() && oldStat.Mode()&syscall.S_IFDIR != syscall.S_IFDIR) ||
-				!sameFsTimeSpec(oldStat.Mtim(), newStat.Mtim()) ||
+				(oldStat.Mode()&syscall.S_IFDIR != syscall.S_IFDIR &&
```

Comment:
Ah, because the line above wasnt comparing to zero... :-)
On Mar 16, 2015 4:55 PM, "Alexander Morozov" <notifications@github.com>
wrote:

> In pkg/archive/changes.go
> <https://github.com/docker/docker/pull/11422#discussion_r26526731>:
>
> > @@ -220,8 +220,8 @@ func (info *FileInfo) addChanges(oldInfo *FileInfo, changes *[]Change) {
> >  				oldStat.Gid() != newStat.Gid() ||
> >  				oldStat.Rdev() != newStat.Rdev() ||
> >  				// Don't look at size for dirs, its not a good measure of change
> > -				(oldStat.Size() != newStat.Size() && oldStat.Mode()&syscall.S_IFDIR != syscall.S_IFDIR) ||
> > -				!sameFsTimeSpec(oldStat.Mtim(), newStat.Mtim()) ||
> > +				(oldStat.Mode()&syscall.S_IFDIR != syscall.S_IFDIR &&
>
> Why we comparing not with 0? Everywhere in other code we comparing with
> it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/11422/files#r26526731>.
>